src/main/native/pdhmm/pdhmm.h

```diff
@@ -0,0 +1,1131 @@
+}
+
+inline void freeVec(double *matchMatrixVec, double *insertionMatrixVec, double *deletionMatrixVec, double *branchMatchMatrixVec, double *branchInsertionMatrixVec, double *branchDeletionMatrixVec, double *transitionVec, double *priorVec, bool *&constantsAreInitialized, bool *&initialized, INT_TYPE *&prev_hap_bases_length)
+{
+	_mm_free(matchMatrixVec);
+	_mm_free(insertionMatrixVec);
+	_mm_free(deletionMatrixVec);
+	_mm_free(branchMatchMatrixVec);
+	_mm_free(branchInsertionMatrixVec);
+	_mm_free(branchDeletionMatrixVec);
+	_mm_free(transitionVec);
+	_mm_free(priorVec);
+	_mm_free(constantsAreInitialized);
+	_mm_free(initialized);
+	_mm_free(prev_hap_bases_length);
+}
+
+inline void initializeVec(double *matchMatrixVec, double *insertionMatrixVec, double *deletionMatrixVec, double *branchMatchMatrixVec, double *branchInsertionMatrixVec, double *branchDeletionMatrixVec, double *transitionVec, double *priorVec, bool &constantsAreInitialized, bool &initialized, const int32_t maxReadLength, const int32_t maxHaplotypeLength)
+{
+	const int32_t paddedMaxReadLength = maxReadLength + 1;
+	const int32_t paddedMaxHaplotypeLength = maxHaplotypeLength + 1;
+	int32_t sizeOfTables = paddedMaxHaplotypeLength * SIMD_WIDTH_DOUBLE * sizeof(double);
+
+	for (int32_t i = 0; i < paddedMaxHaplotypeLength * SIMD_WIDTH_DOUBLE; i++)
+	{
+		matchMatrixVec[i] = 0;
+		insertionMatrixVec[i] = 0;
+		deletionMatrixVec[i] = 0;
+		branchMatchMatrixVec[i] = 0;
+		branchInsertionMatrixVec[i] = 0;
+		branchDeletionMatrixVec[i] = 0;
+	}
+
+	constantsAreInitialized = false;
+	initialized = true;
+}
+
+inline VEC_DOUBLE_TYPE qualToErrorProb(VEC_INT_TYPE qual, const double *qualToErrorProbCache)
+{
+	VEC_INT_TYPE vindex = VEC_AND_INT(qual, VEC_SET1_INT(0xff));
+	VEC_DOUBLE_TYPE result = VEC_GATHER_PD(vindex, qualToErrorProbCache, 8);
+	return result;
+}
+
+inline VEC_DOUBLE_TYPE qualToProb(VEC_INT_TYPE qual, const double *qualToErrorProbCache)
+{
+	return VEC_SUB_PD(VEC_SET1_PD(1), qualToErrorProb(qual, qualToErrorProbCache));
+}
+
+inline VEC_DOUBLE_TYPE matchToMatchProbFun(VEC_INT_TYPE insQual, VEC_INT_TYPE delQual, const double *matchToMatchProb, int32_t &status)
+{
+	VEC_INT_TYPE minQual = VEC_MIN_INT(insQual, delQual);
+	VEC_INT_TYPE maxQual = VEC_MAX_INT(insQual, delQual);
+
+	VEC_MASK_TYPE ltzero = VEC_CMP_LT_INT(minQual, VEC_SET0_INT());
+	if (VEC_CMP_NE_MASK(ltzero, 0))
+	{
+		status = PDHMM_INPUT_DATA_ERROR;
+		return VEC_SET1_PD(-1.0);
+	}
+
+	VEC_MASK_TYPE ltMaxQual = VEC_CMP_LT_INT(VEC_SET1_INT(MAX_QUAL), maxQual);
+
+	// True value
+	VEC_DOUBLE_TYPE a = VEC_SET1_PD(-0.1);
+	VEC_DOUBLE_TYPE trueVal = approximateLog10SumLog10(VEC_MUL_PD(VEC_CVT_INT_PD(minQual), a), VEC_MUL_PD(VEC_CVT_INT_PD(maxQual), a));
+	trueVal = pow_pd(10.0, trueVal, SIMD_WIDTH_DOUBLE, status);
+	if (status != PDHMM_SUCCESS)
+		return VEC_SET1_PD(-1.0);
+	a = VEC_SET1_PD(1.0);
+	trueVal = VEC_SUB_PD(a, trueVal);
+
+	// False value
+
+	// index = [((maxQual * (maxQual + 1)) >> 1) + minQual
```

Comment:
Is this a commented out code or an explanation of code below? If the former then please delete it and if the latter then add a note making it clearer.